src/utils/config.py

```diff
@@ -48,3 +58,31 @@ def init_config():
 
 
 config = init_config()
+
+
+def get_elasticsearch_auth_header():
+    """
+    Build the Elasticsearch Authorization header based on available credentials.
+    Returns None if no credentials are configured.
+
+    Priority:
+    1. If username and password are set, use Basic authentication
+    2. If only auth token is set, use Bearer authentication
```

Comment:
Why this precedence?

Reply:
> Equally easy to revoke (just disable that one user/key)

I don't think that's true, it's easier to revoke a token than change a password

Reply:
How is it easier? Both will require shutting down and restarting the service?

Reply:
I assumed it was easier on the ES side, like how in KBase revoking a token is trivial.

If that's not the case and you're saying that passwords and tokens are completely indistinguishable for this use case then we should just drop token support

Reply:
I don't know, I haven't tested it out, I was thinking more from the side of what happens to the service. Ok I will remove it.